src/textual/widgets/_option_list.py

```diff
@@ -647,6 +655,65 @@ def remove_option_at_index(self, index: int) -> Self:
             ) from None
         return self
 
+    def _replace_option_prompt(self, index: int, prompt: RenderableType) -> None:
+        """Replace the prompt of an option in the list.
+
+        Args:
+            index: The index of the option to replace the prompt of.
+            prompt: The new prompt for the option.
+
+        Raises:
+            IndexError: If there is no option of the given index.
+        """
+        option = self._options[index]
```

Comment:
There is a get_option_at_index() function already, could we use that here instead for consistency?

```suggestion
        option = self.get_option_at_index(index)
```

Reply:
That's a good idea, for sure.

Reply:
Good catch.

Reply:
This PR is really good, implements the feature required in the issue very well.

I feel like we can do better to avoid rewriting code that already exists here. What do you think @nekeal ?

Reply:
@BenMarkFrost  I pushed the changes. Should I replace it also in the `_remove_option` method?

Reply:
Which part of the code do you mean?

Reply:
> @BenMarkFrost I pushed the changes. Should I replace it also in the `_remove_option` method?

Yes well spotted, I think that would be a good idea. I can also see it in `_set_option_disabled()`




> Which part of the code do you mean? The one that retrieves the option's index?

This PR re-implements errors for retrieving options by index and ID, when they are already present in those dedicated functions:

`get_option()` (which gets the option by option_id), and `get_option_by_index()`.

------

The implementation in my opinion could be as simple as

```sh
def replace_option_prompt_at_index(index: int .......):
    self.get_option_at_index(index).set_prompt(new_prompt)
    
def replace_option_prompt(id: str .......):
    self.get_option(id).set_prompt(new_prompt)
```

What do you think?

Reply:
Sounds good to me :rocket: 

Should I also change implementations of other methods so they use `get_option()` and `get_option_by_index()`.

Reply:
Only if you have time / feel like it would be in scope, I think we can happily create another issue for that if it's too much work!

Reply:
I just pushed a [commit](https://github.com/Textualize/textual/pull/2985/commits/1c9f079b5da2c1658e2056d485307fa66f25563b) which shows how the rest of the methods can be implemented. Can you have a look?